deps/v8/src/base/platform/platform-solaris.cc

```diff
@@ -70,7 +70,7 @@ Stack::StackSlot Stack::GetStackStart() {
   pthread_attr_t attr;
   int error;
   pthread_attr_init(&attr);
-  error = pthread_attr_get_np(pthread_self(), &attr);
+  error = 0;
   if (!error) {
```

Comment:
Ideally, of course, we should find another way to check if there was an error in getting the `np` on Solaris (when `pthread_attr_get_np` seems not to exist on Solaris).

Reply:
> `pthread_attr_get_np` doesn't `get` the value of `np`, the ["np" stands for "non portable"](https://stackoverflow.com/questions/2238564/pthread-functions-np-suffix) meaning that it's a platform specific extension.

OK, whatever. It clearly checks if some error exists so that the program can act accordingly. Pretending the error does not exist is not a good strategy, is it?

Reply:
> > `pthread_attr_get_np` doesn't `get` the value of `np`, the ["np" stands for "non portable"](https://stackoverflow.com/questions/2238564/pthread-functions-np-suffix) meaning that it's a platform specific extension.
> 
> OK, whatever. It clearly checks if some error exists so that the program can act accordingly. Pretending the error does not exist is not a good strategy, is it?

To check for errors you can get the returned value from `thr_stksegment`. Documentation for that function is on: https://docs.oracle.com/cd/E23824_01/html/821-1465/thr-stksegment-3c.html#scrolltoc
So basically you can do the same error checking as before:
```
int error;
...
error = thr_stksegment(&s);
if (!error) {
    return s.ss_sp;
}
return nulltpr;
```

Reply:
> > > `pthread_attr_get_np` doesn't `get` the value of `np`, the ["np" stands for "non portable"](https://stackoverflow.com/questions/2238564/pthread-functions-np-suffix) meaning that it's a platform specific extension.
> > 
> > 
> > OK, whatever. It clearly checks if some error exists so that the program can act accordingly. Pretending the error does not exist is not a good strategy, is it?
> 
> To check for errors you can get the returned value from `thr_stksegment`. Documentation for that function is on: https://docs.oracle.com/cd/E23824_01/html/821-1465/thr-stksegment-3c.html#scrolltoc
> So basically you can do the same error checking as before:
> 
> ```
> int error;
> ...
> error = thr_stksegment(&s);
> if (!error) {
>     return s.ss_sp;
> }
> return nullptr;
> ```

Thank you, useful information. However, why would `return nullptr` be the correct response? Wouldn't throwing some exception be more appropriate?